lib/src/barback/dartdevc/dartdevc_module_transformer.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:barback/barback.dart';
+import 'package:bazel_worker/bazel_worker.dart';
+import 'package:path/path.dart' as p;
+
+import '../../io.dart';
+import 'workers.dart';
+import 'module.dart';
+import 'module_reader.dart';
+import 'temp_environment.dart';
+import 'linked_summary_transformer.dart';
+
+/// Creates dartdevc modules given [moduleConfigName] files which describe a set
+/// of [Module]s.
+///
+/// Linked summaries should have already been created for all modules and will
+/// be read in to create the modules.
+class DartDevcModuleTransformer extends Transformer {
+  @override
+  String get allowedExtensions => moduleConfigName;
+
+  DartDevcModuleTransformer();
+
+  @override
+  Future apply(Transform transform) async {
+    TempEnvironment tempEnv;
+    var reader = new ModuleReader(transform.readInputAsString);
+    var modules = await reader.readModules(transform.primaryInput.id);
+    try {
+      var allAssetIds = new Set<AssetId>();
+      var summariesForModule = <ModuleId, Set<AssetId>>{};
+      for (var module in modules) {
+        var transitiveModuleDeps = await reader.readTransitiveDeps(module);
+        var linkedSummaryIds = transitiveModuleDeps.map((depId) {
+          assert(depId.name.isNotEmpty);
+          var summaryDir = depId.name.split('__').first;
+          return new AssetId(depId.package,
+              p.join(summaryDir, '${depId.name}$linkedSummaryExtension'));
+        }).toSet();
+        summariesForModule[module.id] = linkedSummaryIds;
+        allAssetIds..addAll(module.assetIds)..addAll(linkedSummaryIds);
+      }
+      // Create a single temp environment for all the modules in this package.
+      tempEnv = await TempEnvironment.create(allAssetIds, transform.readInput);
+      var outputDir = topLevelDir(transform.primaryInput.id.path);
+      await Future.wait(modules.map((m) => _createDartdevcModule(
+          m, outputDir, tempEnv, summariesForModule[m.id], transform)));
+    } finally {
+      tempEnv?.delete();
+    }
+  }
+}
+
+/// Compiles [module] using the `dartdevc` binary from the SDK to a relative
+/// path under the package that looks like `$outputDir/${module.id.name}.js`.
+Future _createDartdevcModule(
+    Module module,
+    String outputDir,
+    TempEnvironment tempEnv,
+    Set<AssetId> linkedSummaryIds,
+    Transform transform) async {
+  var logger = transform.logger;
```

Comment:
This local variable doesn't add a lot of value. Maybe just call `transform.logger` directly below?